src/app/CommandResponseSender.cpp

```diff
@@ -38,12 +38,10 @@ CHIP_ERROR CommandResponseSender::OnMessageReceived(Messaging::ExchangeContext *
         err = statusError;
         VerifyOrExit(err == CHIP_NO_ERROR, failureStatusToSend.SetValue(Status::InvalidAction));
 
-        // If SendCommandResponse() fails, we are responsible for closing the exchange,
-        // as stipulated by the API contract. We fulfill this obligation by not sending
-        // a message expecting a response on the exchange. Since we are in the middle
-        // of processing an incoming message, the exchange will close itself once we are
-        // done processing it, if there is no response to wait for at that point.
         err = SendCommandResponse();
+        // If SendCommandResponse() fails, we must still close the exchange. Since sending an
+        // InvokeResponseMessage seems problematic, we'll send a StatusResponse with 'Failure'
+        // instead.
```

Comment:
This lost the key part of the comment that was deleted, which is that if we don't send a message needing a response, that closes the exchange.

We could also send absolutely nothing and that would _also_ close the exchange.

So the comment no longer really explains why the code is doing what it's doing.  We need to fix this part.

1.  How are we ensuring the exchange is closed?  Just but not sending an "expect a reply" response.  This works because we are inside OnMessageReceived.
2. Why are we sending anything at all?